sdks/python/apache_beam/dataframe/expressions.py

```diff
@@ -24,60 +30,77 @@ class Session(object):
   def __init__(self, bindings={}):
     self._bindings = dict(bindings)
 
-  def evaluate(self, expr):
+  def evaluate(self, expr):  # type: (Session, Expression) -> Any
     if expr not in self._bindings:
       self._bindings[expr] = expr.evaluate_at(self)
     return self._bindings[expr]
 
-  def lookup(self, expr):
+  def lookup(self, expr):  #  type: (Session, Expression) -> Any
     return self._bindings[expr]
 
 
+# The return type of an Expression
+T = TypeVar('T')
+
+
 class Expression(object):
   """An expression is an operation bound to a set of arguments.
 
   An expression represents a deferred tree of operations, which can be
   evaluated at a specific bindings of root expressions to values.
   """
-  def __init__(self, name, proxy=None, _id=None):
+  def __init__(
+      self,
+      name,  # type: str
+      proxy,  # type: T
+      _id=None  # type: Optional[str]
+  ):
     self._name = name
     self._proxy = proxy
     # Store for preservation through pickling.
     self._id = _id or '%s_%s' % (name, id(self))
 
-  def proxy(self):
+  def proxy(self):  # type: (Expression) -> T
     return self._proxy
 
   def __hash__(self):
     return hash(self._id)
 
-  def __eq__(self):
+  def __eq__(self, other):
     return self._id == other._id
 
   def __ne__(self, other):
     return not self == other
 
-  def evaluate_at(self, session):
+  def evaluate_at(self, session):  # type: (Expression, Session) -> T
     """Returns the result of self with the bindings given in session."""
     raise NotImplementedError(type(self))
 
-  def requires_partition_by_index(self):
+  def requires_partition_by_index(self):  # type: (Expression) -> bool
     """Whether this expression requires its argument(s) to be partitioned
     by index."""
     # TODO: It might be necessary to support partitioning by part of the index,
     # for some args, which would require returning more than a boolean here.
     raise NotImplementedError(type(self))
 
-  def preserves_partition_by_index(self):
+  def preserves_partition_by_index(self):  # type: (Expression) -> bool
     """Whether the result of this expression will be partitioned by index
     whenever all of its inputs are partitioned by index."""
     raise NotImplementedError(type(self))
 
 
 class PlaceholderExpression(Expression):
   """An expression whose value must be explicitly bound in the session."""
-  def __init__(self, proxy):
+  def __init__(
+      self,  # type: PlaceholderExpression
+      proxy  # type: T
+  ):
     super(PlaceholderExpression, self).__init__('placeholder', proxy)
+    """Initialize a placeholder expression.
+
+    Args:
+      proxy: A proxy object with the type expected to be bound to this expression. Used for type checking at pipeline construction time.
```

Comment:
Long line.

Reply:
Fixed (here and elsewhere). I tried to run yapf on this, but it seems to not care about long lines in docstrings? I just wrapped the lines manually for now.

Reply:
Huh. Maybe it figures adding newlines in string literals could introduce semantic changes to the code.